File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit=0, bestBuy=INT_MAX;
        for(int p:prices){
            bestBuy=min(bestBuy,p);
            maxProfit=max(maxProfit,p-bestBuy);
        }
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProfit=0
        bestBuy=prices[0]
        for price in prices:
            bestBuy=min(bestBuy,price)
            maxProfit=max(maxProfit, price-bestBuy)
        return maxProfit